src/library/autodj/autodjprocessor.cpp

```diff
@@ -1127,12 +1144,28 @@ double AutoDJProcessor::getOutroStartSecond(DeckAttributes* pDeck) {
 
 double AutoDJProcessor::getOutroEndSecond(DeckAttributes* pDeck) {
     const mixxx::audio::FramePos trackEndPosition = pDeck->trackEndPosition();
+    const mixxx::audio::FramePos outroStartPosition = pDeck->outroStartPosition();
     const mixxx::audio::FramePos outroEndPosition = pDeck->outroEndPosition();
     if (!outroEndPosition.isValid() || outroEndPosition > trackEndPosition) {
-        return getLastSoundSecond(pDeck);
+        double lastSoundSecond = getLastSoundSecond(pDeck);
+        if (!outroStartPosition.isValid() || outroStartPosition > trackEndPosition) {
+            // No outro start and outro end set, use Last Sound.
+            return lastSoundSecond;
+        }
+        double outroStartSecond = framePositionToSeconds(outroStartPosition, pDeck);
+        if (m_transitionTime >= 0 && lastSoundSecond > outroStartSecond) {
+            double outroEndFromTime = outroStartSecond + m_transitionTime;
+            if (outroEndFromTime < lastSoundSecond) {
```

Comment:
you'll also need to check for `outroEndFromTime < trackEndPosition` IIUC.

Reply:
lastSoundSecond has been clamped to be <= trackEndPosition
I will add a note.